code/game/machinery/machinery.dm

```diff
@@ -161,6 +161,8 @@ Class Procs:
 	var/obj/item/weapon/card/id/scan = null	//ID inserted for identification, if applicable
 	var/id_tag = null // Identify the machine
 
+	var/area/power_area //The area the machine is drawing power from.
```

Comment:
You seem to always be writing to this the variable before using it, so I don't see the point of it being here.